crypto/dsa/dsa_ossl.c

```diff
@@ -163,6 +163,10 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
     } else
         ctx = ctx_in;
 
+    /* Preallocate space for k */
+    if (!BN_set_bit(k, BN_num_bits(dsa->q) + 1))
```

Comment:
It's kind of unexpected to see +1 here. I mean number of corresponding length would be -1... Not saying that it's wrong though...

Reply:
We're wanting one more bit that dsa->q has.  The end result is a constant that is guaranteed one bit longer.

I do agree that this approach is less clear than what I used for ECDSA (two adds) but it will be faster.